specs/dialog/dialog.md

```diff
@@ -0,0 +1,125 @@
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/dialogs)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-dialog`
+
+*Attributes:*
+- `modal` - boolean value for modal functionality. Sets aria-modal to true (defaults to true).
+- `hidden` - boolean value to toggle the visibility of the dialog (defaults to false).
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*CSS custom properties:*
+- width
+- height
+
+*Slots:*
+- default slot for content
+
+*Events*
+- dismiss - event fired when a user presses escape or clicks outside the dialog content
+
+### Anatomy and Appearance
+**Structure:**
+```
+<div class="root">
+    <div class="positioning-region">
+        <div class="overlay">
+        <div
+            role="dialog"
+            aria-modal="true"
+            tabindex="-1"
+            class="content-region"
+            aria-describedby="foo"
+            aria-labelledby="bar"
+            aria-label="baz"
+        >
+            <slot></slot>
+        </div>
+    </div>
+</div>
+```
+
+**Appearance:**
+| State | Image |
+| ----- | ----- |
+| default | ![](./images/dialog.png) |
+| dark mode | ![](./images/dialog-dark.png) |
+
+Parts:
+- root - the root of the dialog
+- positioning-region - ensuring the dialog is centered correctly requires a certain structure. Centering with absolute positioning can cause blurry content within the dialg itself. After investigating several implementations, the best and most common way of ensuring the dialog can be centered is to include a div to position the actual dialog itself using something like flexbox or css-grid. This ensures that the blurry content issue does not happen and the dialog centers easily within the screen.
+- modal-overlay - the modal overlay
+- content-region - the region where content is actually rendered. This part is where the `role="dialog"` will actually exist
+
+Animation:
+The current working model (assumption) is that animation iwill be taken up as part of the adaptive ui story in the design system at some point. This will provide configurability to users. From a dialog standpoint, I think we'll have a default animation baked in, though this would be configurable through the design system.
+
+If we compose our dialog with two sets of styles, one for the dialog itself and another for the animation, that would also provide an easy way for users to create a new instance:
+
+```js
+@customElement({
+    name: "my-dialog",
+    template,
+    styles: css`${dialogBaseStyles}${myAnimationStyles}`
+})
+```
+---
+
+## Implementation
+
+### States
+
+Visible - The dialog is visible. This state is managed solely by the app author via the visible attribute.
```

Comment:
Need to update this to align with the note on `hidden` above.

Reply:
Ruh-roh...fixed! Thanks!